src/main/cpp/util/file_windows.cc

```diff
@@ -534,7 +535,10 @@ bool CanAccessDirectory(const Path& path) {
 
   // The only easy way to know if a directory is writable is by attempting to
   // open a file for writing in it.
-  Path dummy_path = path.GetRelative("bazel_directory_access_test");
+  std::string dummy_name = "bazel_directory_access_test_";
+  dummy_name += std::to_string(::GetCurrentThreadId());
```

Comment:
Please add a brief comment here saying that the Thread ID avoids races among concurrent Bazel processes.

Reply:
@laszlocsomor can you restart checks? They are failed on platforms that I didn't change.